### PR TITLE
fix: Migrate AWS Cloudwatch alert channel to API v2

### PR DESCRIPTION
**Pull request by vatasha**
---
Migrate cloudwatch alert channel to API v2
fix(api):  Migrate cloudwatch alert channel to API v2
---

***Issue***: https://lacework.atlassian.net/jira/software/projects/ALLY/boards/58?selectedIssue=ALLY-637

### CODE REVIEW

File: lacework/resource_lacework_alert_channel_aws_cloudwatch.go [hunk 1]
```diff
@@ -180,84 +173,41 @@ func resourceLaceworkAlertChannelAwsCloudWatchUpdate(d *schema.ResourceData, met
 	alert.IntgGuid = d.Id()
 
 	log.Printf("[INFO] Updating %s integration with data:\n%+v\n", api.AwsCloudWatchIntegration, alert)
```
Comment (dmurray-lacework) [inline lacework/resource_lacework_alert_channel_aws_cloudwatch.go:175]:
```suggestion
	log.Printf("[INFO] Updating %s integration with data:\n%+v\n", api.CloudwatchEbAlertChannelType, alert)
```

File: lacework/resource_lacework_alert_channel_aws_cloudwatch.go [hunk 2]
```diff
@@ -96,77 +102,64 @@ func resourceLaceworkAlertChannelAwsCloudWatchCreate(d *schema.ResourceData, met
...
 
-	log.Printf("[INFO] Created %s integration with guid %s\n", api.AwsCloudWatchIntegration, integration.IntgGuid)
+	log.Printf("[INFO] Created %s integration with guid %s\n", api.CloudwatchEbAlertChannelType, integration.IntgGuid)
 	return nil
 }
 
 func resourceLaceworkAlertChannelAwsCloudWatchRead(d *schema.ResourceData, meta interface{}) error {
 	lacework := meta.(*api.Client)
 
 	log.Printf("[INFO] Reading %s integration with guid %s\n", api.AwsCloudWatchIntegration, d.Id())
```
Comment (dmurray-lacework) [inline lacework/resource_lacework_alert_channel_aws_cloudwatch.go:135]:
```suggestion
	log.Printf("[INFO] Reading %s integration with guid %s\n", api.CloudwatchEbAlertChannelType, d.Id())
```

File: lacework/resource_lacework_alert_channel_aws_cloudwatch.go [hunk 3]
```diff
@@ -22,21 +22,25 @@ func resourceLaceworkAlertChannelAwsCloudWatch() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"name": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The integration name",
 			},
 			"intg_guid": {
```
Comment (dmurray-lacework) [inline lacework/resource_lacework_alert_channel_aws_cloudwatch.go:29]: nit: group computed values together below values requiring input

File: lacework/resource_lacework_alert_channel_aws_cloudwatch.go (hunk 1 above)
Comment (vatasha) [reply lacework/resource_lacework_alert_channel_aws_cloudwatch.go:175]: thanks for catching this!

File: lacework/resource_lacework_alert_channel_aws_cloudwatch.go (hunk 2 above)
Comment (vatasha) [reply lacework/resource_lacework_alert_channel_aws_cloudwatch.go:135]: thanks for catching this!

**Review by dmurray-lacework — APPROVED** [review]
Just a couple of suggestions, but Looks good!

File: lacework/resource_lacework_alert_channel_aws_cloudwatch.go [hunk 4]
```diff
@@ -22,21 +22,20 @@ func resourceLaceworkAlertChannelAwsCloudWatch() *schema.Resource {
...
+				Optional:    true,
+				Default:     true,
+				Description: "The state of the external integration",
 			},
 			"event_bus_arn": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ARN of your AWS CloudWatch event bus",
```
Comment (afiune) [inline lacework/resource_lacework_alert_channel_aws_cloudwatch.go:38]: `CloudWatch` or `Cloudwatch`? 😅 just kidding - this is acceptable 😉

**Review by afiune — APPROVED** [review]
AWESOME!!

tenor-213853359

**Pull request merged**